addons/pos_cache/models/pos_cache.py

```diff
@@ -94,7 +93,7 @@ def get_products_from_cache(self, fields, domain):
             new_cache = self._get_cache_for_user()
             return new_cache.get_cache(domain, fields)
 
-    @api.one
     def delete_cache(self):
-        # throw away the old caches
-        self.cache_ids.unlink()
+        for cache in self:
+            # throw away the old caches
+            cache.cache_ids.unlink()
```

Comment:
You can actually simplify to
```python
self.cache_ids.unlink()
```
because `self.cache_ids` behaves as `mapped()` if `self` contains several records.